components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/importexport/utils/APIImportUtil.java

```diff
@@ -348,28 +367,32 @@ public static void importAPI(String pathToArchive, String currentUser, boolean i
             if (!APIConstants.APITransportType.WS.toString().equalsIgnoreCase(importedApi.getType())) {
                 String swaggerContent = loadSwaggerFile(pathToArchive);
                 addSwaggerDefinition(importedApi.getId(), swaggerContent, apiProvider);
-
-                //Load required properties from swagger to the API
-                APIDefinition apiDefinition = OASParserUtil.getOASParser(swaggerContent);
-                Set<URITemplate> uriTemplates = apiDefinition.getURITemplates(swaggerContent);
-                for (URITemplate uriTemplate : uriTemplates) {
-                    Scope scope = uriTemplate.getScope();
-                    if (scope != null && !(APIUtil.isWhiteListedScope(scope.getKey()))
-                            && apiProvider.isScopeKeyAssigned(importedApi.getId(), scope.getKey(), tenantId)) {
-                        String errorMessage =
-                                "Error in adding API. Scope " + scope.getKey() + " is already assigned by another API.";
-                        log.error(errorMessage);
-                        throw new APIImportExportException(errorMessage);
+                //If graphQL API, import graphQL schema definition to registry
+                if (StringUtils.equals(importedApi.getType(), APIConstants.APITransportType.GRAPHQL.toString())) {
+                    String schemaDefinition = loadGraphqlSDLFile(pathToArchive);
+                    addGraphqlSchemaDefinition(importedApi, schemaDefinition, apiProvider);
+                } else {
+                    //Load required properties from swagger to the API
+                    APIDefinition apiDefinition = OASParserUtil.getOASParser(swaggerContent);
+                    Set<URITemplate> uriTemplates = apiDefinition.getURITemplates(swaggerContent);
+                    for (URITemplate uriTemplate : uriTemplates) {
```

Comment:
Can't we use API object to retrieve the operations instead of retriving them from swagger Definition? All the details of graphql operations (scope, throttle policy, security) are availble at the api Object. So, you can use the same function **importedApi.setUriTemplates(uriTemplates)**  for the graphql APIs as well.

Reply:
The limitation that we cannot use API object (api.json) to add the scopes and URI templates for the other general REST APIs, is that the URI Templates are not properly populated in the API object when exporting in cases where there are duplicate resource paths used with different Http verbs. Hence we are using the swagger definition to add the URI templates and scopes for non-graphql and non-WebSocket APIs.

For graphql APIs, we are getting the URI templates from the API object (api.json). When exporting a graphql API, the URI Templates are not set to empty.

Reply:
Yes. So,  **importedApi.setUriTemplates(uriTemplates)**
Has this functionality been used in Graphql APIs as well? In here I can only see the logic in else part ?

Reply:
The API is created before that if from https://github.com/wso2/carbon-apimgt/pull/7159/files/c08b8adf44e0e31fe91b84650c643a563a57febf#diff-55799393a1536cac0ba98feefe7a15a5L344

Since graphql APIs already have URI templates in the API object, those are not needed to be set again. when adding the API, URI templates will be added to the database in the API creation flow